bdk-ffi/src/esplora.rs

```diff
@@ -81,4 +84,14 @@ impl EsploraClient {
             .broadcast(&bdk_transaction)
             .map_err(EsploraError::from)
     }
+
+    pub fn get_tx(&self, txid: String) -> Result<Option<Arc<Transaction>>, EsploraError> {
+        let txid = Txid::from_str(txid.as_str()).map_err(|e| EsploraError::Parsing {
+            error_message: format!("Invalid txid: {}", e),
+        })?;
```

Comment:
Do I understand correctly that this is sort of remapping the error you might get from the `Txid::from_str()` into an `EsploraError::Parsing`?

I do think it probably makes sense here because otherwise you'd need to combine both the Txid parsing error and the Esplora error into one, which gets messy when there are multiple of those combinations across the codebase.

Reply:
Other observation: with the introduction of the bitcoin-ffi type `Txid` and the `impl_string_custom_typedef` macro, can that translation into the Rust Txid type happen automatically? It's a String on the target language side but becomes a true Rust `Txid` when you feed it to a Rust function? It's my first time using the TypeDef in practice so not 100% sure... but maybe?

Reply:
> Do I understand correctly that this is sort of remapping the error you might get from the `Txid::from_str()` into an `EsploraError::Parsing`?
> 
> I do think it probably makes sense here because otherwise you'd need to combine both the Txid parsing error and the Esplora error into one, which gets messy when there are multiple of those combinations across the codebase.

yup that's correct.

Needing to remap the error that we get back from parsing which is different than the `EsploraError` type that we return from the function.

Looking at the Txid docs for `from_str` it seemed like the `EsploraError::Parsing` would make the most sense.

But yeah I played around with a few things, and what you mention about combining seemed a bit messy to me too.

The one thing I did think about doing was creating a conversion for it in `error.rs` like:

```
impl From<BdkHexToArrayError> for EsploraError {
    fn from(error: BdkHexToArrayError) -> Self {
        EsploraError::Parsing {
            error_message: format!("Invalid txid: {}", error),
        }
    }
}
```

but `bitcoin::hashes::hex::HexToArrayError` would need to be added as a dependency so I didn't think it was worth adding a new dependency based on one error to be able to clean up 

```
     let txid = Txid::from_str(txid.as_str()).map_err(|e| EsploraError::Parsing {
            error_message: format!("Invalid txid: {}", e),
        })?;
```

to something like `let txid = Txid::from_str(&txid)?;`.

but as I'm writing this it's dawning on me that bdk_wallet probably exports that dependency so I dont need to explicitly add it which solves my initial hesitation to add the conversion of the error in `error.rs`.... so look out for a push with that change right now to make things look even nicer.

Reply:
> Other observation: with the introduction of the bitcoin-ffi type `Txid` and the `impl_string_custom_typedef` macro, can that translation into the Rust Txid type happen automatically? It's a String on the target language side but becomes a true Rust `Txid` when you feed it to a Rust function? It's my first time using the TypeDef in practice so not 100% sure... but maybe?

From my understanding that `impl_string_custom_typedef` macro only helps with conversions at the FFI boundary not within Rust functions.